File: backend/app/schemas/governance.py
from datetime import datetime

from pydantic import BaseModel


class GovernanceSection(BaseModel):
    status: str
    source: str
    count: int | None = None
    detail: str | None = None


class GovernanceSummary(BaseModel):
    generated_at: datetime
    ai: GovernanceSection
    tasks: GovernanceSection
    statistics: GovernanceSection
    report: GovernanceSection
    settings: GovernanceSection

File: src/lib/api/governance.ts
import { apiFetch } from './client'

export interface GovernanceSection {
  status: 'ready' | 'empty' | 'unavailable'
  source: string
  count: number | null
  detail: string | null
}

export interface GovernanceSummary {
  generated_at: string
  ai: GovernanceSection
  tasks: GovernanceSection
  statistics: GovernanceSection
  report: GovernanceSection
  settings: GovernanceSection
}

export function getGovernanceSummary(): Promise<GovernanceSummary> {
  return apiFetch<GovernanceSummary>('/api/admin/governance/summary')
}
